ostd/src/arch/riscv/iommu/mod.rs

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! The IOMMU support.
+
+use crate::mm::{dma::Daddr, Paddr};
+
+/// An enumeration representing possible errors related to IOMMU.
+#[derive(Debug)]
+pub enum IommuError {
+    /// No IOMMU is available.
+    NoIommu,
+}
+
+///
+/// # Safety
+///
+/// Mapping an incorrect address may lead to a kernel data leak.
+pub(crate) unsafe fn map(_daddr: Daddr, _paddr: Paddr) -> Result<(), IommuError> {
+    Err(IommuError::NoIommu)
+}
+
+pub(crate) fn unmap(_daddr: Daddr) -> Result<(), IommuError> {
+    Err(IommuError::NoIommu)
+}
+
+pub(crate) fn init() -> Result<(), IommuError> {
+    Err(IommuError::NoIommu)
```

Comment:
The framekernel architecture requires IOMMU or something equivalent to prevent malicious or mistaken DMA that corrupts typed memory. So I googled "RISC-V IOMMU"; there is indeed a RISC-V IOMMU specification. We should and will implement IOMMU in the future. So I suggest adding a comment to state this intention.

```rust
pub(crate) fn init() -> Result<(), IommuError> {
    // TODO: We will support IOMMU on RISC-V
    Err(IommuError::NoIommu)
}
```